aiida-registry-app/src/Components/MainIndex.jsx

```diff
@@ -61,12 +65,12 @@ function preparePluginsForSearch(plugins) {
 const pluginsListForSearch = preparePluginsForSearch(plugins);
 
 function Search() {
-  const { searchQuery, setSearchQuery, sortedData, setSortedData } = useSearchContext();
+  const { searchQuery, setSearchQuery, setSearchResults, isSearchSubmitted, setIsSearchSubmitted } = useSearchContext();
   // Update searchQuery when input changes
   const handleSearch = (searchQuery) => {
     setSearchQuery(searchQuery);
-    if (searchQuery == "") {
-      setSortedData(plugins)
+    if (searchQuery == "" || isSearchSubmitted == true) {
+      setIsSearchSubmitted(false);
```

Comment:
Not sure I understand the second part of condition when `isSearchSubmitted == true` set it to false. Is it used to reset the submit so can accept new query submit?

Reply:
Yes, when you submit a search and then try to modify or remove the search query, the current search will be deleted, and the main index will be displayed until you submit another query or remove the search.

Reply:
What do you think of this? I'm think there is a better way to do this by getting the search query from the URL, This way the search query can be permanent. like `https://aiidateam.github.io/aiida-registry/search?=relaxation`.
Not sure how to implement it but may give it a try later.

Reply:
I think this is good enough. Thanks for the explaination.